### PR TITLE
Some helper API to dynamically work with GLib enums/flags

### PR DESCRIPTION
**Pull request by sdroege**
See commits

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
We have a big lack of documentation in here!

**Comment by sdroege** [comment]
That's very true :) I can write documentation when you tell me that the API is good as-is

File: src/enums.rs [hunk 1]
```diff
@@ -37,3 +41,505 @@ impl ToGlib for UserDirectory {
...
     }
 }
+
+#[derive(Debug)]
+pub struct EnumClass(*mut gobject_ffi::GEnumClass);
+
+impl EnumClass {
+    pub fn new(type_: Type) -> Option<Self> {
+        unsafe {
+            let is_enum: bool = from_glib(gobject_ffi::g_type_is_a(type_.to_glib(), gobject_ffi::G_TYPE_ENUM));
```
Comment (GuillaumeGomez) [inline src/enums.rs]: Why not:

```rust
if from_glib::<bool>(gobject_ffi::g_type_is_a(type_.to_glib(), gobject_ffi::G_TYPE_ENUM)) {
    Some(EnumClass(gobject_ffi::g_type_class_ref(type_.to_glib()) as *mut _))
} else {
    None
}
```

?

File: src/enums.rs [hunk 2]
```diff
@@ -37,3 +41,505 @@ impl ToGlib for UserDirectory {
...
+}
+
+
+#[derive(Debug)]
+pub struct FlagsClass(*mut gobject_ffi::GFlagsClass);
+
+impl FlagsClass {
+    pub fn new(type_: Type) -> Option<Self> {
+        unsafe {
+            let is_flags: bool = from_glib(gobject_ffi::g_type_is_a(type_.to_glib(), gobject_ffi::G_TYPE_FLAGS));
```
Comment (GuillaumeGomez) [inline src/enums.rs]: Same as above.

**Comment by GuillaumeGomez** [comment]
So, it seems good for me as is. My only concern is that I miss something because I don't have idea where to use this code for the moment.

**Comment by EPashkin** [comment]
@sdroege IMHO API is good.

**Comment by sdroege** [comment]
> My only concern is that I miss something because I don't have idea where to use this code for the moment.

https://gist.github.com/sdroege/46868640122e373c814b593652878f79#file-playbin-rs-L24 would be an example of flags usage

**Comment by sdroege** [comment]
Thanks for looking and the very fast feedback, I'll get docs written tomorrow then :)

**Comment by GuillaumeGomez** [comment]
Thanks! :)

**Comment by sdroege** [comment]
Docs added

**Comment by sdroege** [comment]
Anything you want to have added/clarified in the docs or does it look ok?

**Comment by GuillaumeGomez** [comment]
No, that's all good. Since @EPashkin already gave his approval, we can merge. Thanks again!

**Pull request merged**